fdb-extensions/src/main/java/com/apple/foundationdb/linear/RandomMatrixHelpers.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * RandomMatrixHelpers.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.linear;
+
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nonnull;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class RandomMatrixHelpers {
+    private RandomMatrixHelpers() {
+        // nothing
+    }
+
+    @Nonnull
+    public static RealMatrix randomOrthognalMatrix(int seed, int dimension) {
+        return decomposeMatrix(randomGaussianMatrix(seed, dimension, dimension));
+    }
+
+    @Nonnull
+    public static RealMatrix randomGaussianMatrix(int seed, int rowDimension, int columnDimension) {
+        final SecureRandom rng;
+        try {
+            rng = SecureRandom.getInstance("SHA1PRNG");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+        rng.setSeed(seed);
+
+        final double[][] resultMatrix = new double[rowDimension][columnDimension];
+        for (int row = 0; row < rowDimension; row++) {
+            for (int column = 0; column < columnDimension; column++) {
+                resultMatrix[row][column] = nextGaussian(rng);
+            }
+        }
+
+        return new RowMajorRealMatrix(resultMatrix);
+    }
+
+    private static double nextGaussian(@Nonnull final SecureRandom rng) {
+        double v1;
+        double v2;
+        double s;
+        do {
+            v1 = 2 * rng.nextDouble() - 1; // between -1 and 1
+            v2 = 2 * rng.nextDouble() - 1; // between -1 and 1
+            s = v1 * v1 + v2 * v2;
+        } while (s >= 1 || s == 0);
+        double multiplier = StrictMath.sqrt(-2 * StrictMath.log(s) / s);
+        return v1 * multiplier;
+    }
+
+    @Nonnull
+    private static RealMatrix decomposeMatrix(@Nonnull final RealMatrix matrix) {
```

Comment:
It's weird that the name here is about matrix decomposition, but what this returns is just the orthogonal matrix `Q`. In some sense, the right thing from an API perspective would be to also return the triangular matrix `R`

Reply:
I will remove RandomMatrixHelpers entirely. All of this functionality is now embodied in FhtKacRotator. I will also mark every comment re: RandomMatrixHelpers as resolved.